quantumdb-postgresql/src/main/java/io/quantumdb/core/planner/PostgresqlMigrator.java

```diff
@@ -112,18 +112,29 @@ public void applyDataChanges(State state, Stage stage) throws MigrationException
 	}
 
 	@Override
-	public void drop(State state, Version version) throws MigrationException {
+	public void drop(State state, Version version, Stage stage) throws MigrationException {
 		// Check that the version's operation is of type DDL, or has no operation (root version).
 		checkArgument(version.getOperation() == null || version.getOperation().getType() == Type.DDL);
 
 		RefLog refLog = state.getRefLog();
 		Catalog catalog = state.getCatalog();
 
+		Set<Version> activeVersions = refLog.getVersions().stream()
+				.filter(version1 -> !version1.equals(version))
+				.collect(Collectors.toSet());
+
+		// Add latest version of intermediate steps
+		if (stage != null) {
+			activeVersions.add(stage.getLast());
+		}
```

Comment:
The set of active versions is the set that no TableRefs should be deleted from, including the last version that is migrated.